core/src/main/java/org/opensearch/remote/metadata/client/AbstractSdkClient.java

```diff
@@ -168,30 +168,36 @@ protected GetDataObjectResponse getGlobalResourceDataFromCache(GetDataObjectRequ
     }
 
     private GetDataObjectResponse replaceGlobalTenantId(GetDataObjectRequest request, GetDataObjectResponse response) {
-        response.source().put(TENANT_ID_FIELD_KEY, request.tenantId());
         GetResponse getResponse = response.getResponse();
         if (getResponse == null) {
             throw new OpenSearchStatusException(
                 "Cached response is null, please check configuration with system admin!",
                 RestStatus.INTERNAL_SERVER_ERROR
             );
         }
-        try {
-            JsonNode jsonNode = OBJECT_MAPPER.readTree(getResponse.toString());
-            ((ObjectNode) jsonNode.get("_source")).put(TENANT_ID_FIELD_KEY, Optional.ofNullable(request.tenantId()).orElse(DEFAULT_TENANT));
-            XContentParser parser = JsonXContent.jsonXContent.createParser(
-                NamedXContentRegistry.EMPTY,
-                LoggingDeprecationHandler.INSTANCE,
-                OBJECT_MAPPER.writeValueAsString(jsonNode)
-            );
-            return GetDataObjectResponse.builder().id(request.id()).parser(parser).source(response.source()).build();
-        } catch (IOException e) {
-            throw new OpenSearchStatusException(
-                "Failed to parse cached global response, please check configuration with system admin!",
-                RestStatus.INTERNAL_SERVER_ERROR,
-                e
-            );
+        if (response.getResponse().isExists()) {
```

Comment:
We've already extracted `response.getResponse()` into `getResponse` in line 171.  
```suggestion
        if (getResponse.isExists()) {
```
In the case where we don't have the response and we have to construct it from a builder, this could be a performance hit.

Reply:
Addressed here: https://github.com/opensearch-project/opensearch-remote-metadata-sdk/pull/258#discussion_r2379814674